### DISCUSSION TITLE
Access the intended value of a Field of type Literal from uninstanced data model in Pydantic v2

### DISCUSSION THREAD

**Discussion by maralasar** [pydantic/pydantic#8812] [Question]
I have a very specific case where I want to access the value that a Literal field should be assigned without creating an instance of the according data model. I already have a solution in place but feel like it's "wrong" even if it is working.

Let me make an example data model:
```python
from pydantic import BaseModel
from typing import Literal

class Example(BaseModel):
    attr: Literal["Value"]
```

What I want/need to do is have some kind of functionality to get something like
```python
>>> get_literal_value(Example, 'attr')
'Value'
```

Currently I'm doing
```python
def get_literal_value(data_model, attr_name):
    return data_model.model_fields[attr_name].annotation.__args__[0]
```
or alternatively using `get_args`
```python
from typing import get_args

def get_literal_value(data_model, attr_name):
    return get_args(data_model.model_fields[attr_name].annotation)[0]
```

I know this is a very specific application, but I wanted to know if there is intended way to do such a thing?

**Comment by GDYendell** [comment]
I would prefer `get_args()` over `__args__`, because the former is a documented API [1]. You can avoid using either
```python
data_model.__pydantic_core_schema__["schema"]["fields"][attr_name]["schema"]["expected"][0]
```

I am not sure if this is considered a public API or if there is some helper function in pydantic or pydantic_core for accesing schema definitions.

You can also access the annotations independently from pydantic
```python
get_args(data_model.__annotations__[attr_name])[0]
```

I don't know that either of these are necessarily better than your original idea, though. 

[1] - https://stackoverflow.com/a/50101934
